crates/bevy_ui/src/layout/mod.rs

```diff
@@ -5,44 +5,69 @@ use crate::{ContentSize, Node, Style, UiScale};
 use bevy_ecs::{
     change_detection::DetectChanges,
     entity::Entity,
-    event::EventReader,
+    prelude::Component,
     query::{With, Without},
+    reflect::ReflectComponent,
     removal_detection::RemovedComponents,
-    system::{Query, Res, ResMut, Resource},
+    system::{ParamSet, Query, Res, ResMut, Resource},
     world::Ref,
 };
 use bevy_hierarchy::{Children, Parent};
 use bevy_log::warn;
 use bevy_math::Vec2;
+use bevy_reflect::{std_traits::ReflectDefault, Reflect};
 use bevy_transform::components::Transform;
 use bevy_utils::HashMap;
-use bevy_window::{PrimaryWindow, Window, WindowResolution, WindowScaleFactorChanged};
+use bevy_window::Window;
 use std::fmt;
 use taffy::{prelude::Size, style_helpers::TaffyMaxContent, Taffy};
 
+/// Marks an entity as `UI root entity` with an associated root Taffy node and holds the resolution and scale factor information necessary to compute a UI layout.
+#[derive(Component, Debug, Reflect)]
+#[reflect(Component, Default)]
 pub struct LayoutContext {
-    pub scale_factor: f64,
-    pub physical_size: Vec2,
-    pub min_size: f32,
-    pub max_size: f32,
+    /// The size of the root node in the layout tree.
+    ///
+    /// Should match the size of the output window in physical pixels of the display device.
+    pub root_node_size: Vec2,
+    /// [`Style`] properties of UI node entites with `Val::Px` values are multiplied by the `combined_scale_factor` before they are copied to the Taffy layout tree.
+    ///
+    /// `combined_scale_factor` is calculated by multiplying together the `scale_factor` of the output window and [`crate::UiScale`].
+    pub combined_scale_factor: f64,
+    /// After a UI layout has been computed, the layout coordinates are multiplied by `layout_to_logical_factor` to determine the final size of each UI Node entity to be stored in its [`Node`] component.
+    ///
+    /// `layout_to_logical_factor` is the reciprocal of the target window's `scale_factor` and doesn't include [`crate::UiScale`].
+    pub layout_to_logical_factor: f64,
 }
 
-impl LayoutContext {
-    /// create new a [`LayoutContext`] from the window's physical size and scale factor
-    fn new(scale_factor: f64, physical_size: Vec2) -> Self {
+impl Default for LayoutContext {
+    fn default() -> Self {
         Self {
-            scale_factor,
-            physical_size,
-            min_size: physical_size.x.min(physical_size.y),
-            max_size: physical_size.x.max(physical_size.y),
+            root_node_size: Vec2::new(800., 600.),
+            combined_scale_factor: 1.0,
+            layout_to_logical_factor: 1.0,
         }
     }
 }
 
+impl LayoutContext {
+    fn relative_ne(&self, other: &Self) -> bool {
```

Comment:
We generally use `FloatOrd` to compare floats for equality in other places. I don't feel like it's justified to pull a new dependency just for having a slightly shorter syntax in that one function, while adding a new unfamiliar pattern for readers.

Reply:
The new dependency also introduce a duplicate crate version and fails CI.

Reply:
Yeah it's not needed anyway, I was forgetting how floats work, just partial equality is fine here/